ui/src/ui-utils.ts

```diff
@@ -64,3 +64,62 @@ export function truncateText(text: string, limit: number, withEllipsis: boolean
     return text;
   }
 }
+
+/**
+ * Get the correct display name for certain publishers.
+ * 
+ * @param {string} the type of primaryPaperLInk
+ */
+export function getLinkText(input: string): string {
+
+  var linkText: string = "";
+
+  if (input.length > 17) {
+    linkText = "View Via Publisher";
+    return linkText;
+  }
+
+  switch (input) {
+    case "acm":
+      linkText = "View On ACM";
+      break;
```

Comment:
Why not just: 
```suggestion
      return "View On ACM";
```

Could that let you get rid of the `break;`?